sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -1014,3 +1036,58 @@ impl Target for SpiWrapper {
         Rc::clone(&self.underlying_target).assert_cs()
     }
 }
+
+/// Several `I2cWrapper` objects can exist concurrently, sharing the same underlying transport
+/// bus, but having distinct I2C 7-bit address settings.  (Terminology is a little muddy here, in
+/// that the wrapper also implements the I2C "bus" trait, even though it more properly would be
+/// named a "target" or "device".)
+///
+/// Calling e.g. `set_max_speed()` on a `I2cWrapper` will not immediately be propagated to the
+/// underlying transport, instead, the accumulated settings are kept in the `I2cWrapper`, and
+/// propagated only whenever an actual I2C transaction is initiated.
+struct I2cWrapper {
+    /// Reference to the `Bus` instance of the underlying transport.
+    underlying_target: Rc<dyn Bus>,
+    my_default_addr: Cell<Option<u8>>,
+    my_max_speed: Cell<Option<u32>>,
```

Comment:
Stylistic nit: Do we really want to call these `my_...`?

Reply:
Yes, it is somewhat odd.  Generally it is implied that all members of a struct belong to that struct, and "my" is superfluous.

I am trying to recall what I had in mind, when choosing.  Something about two Wrapper objects each keeping track of their own speed settings, etc., and an emphasis that these fields may not be what the underlying hardware is set at, at any point in time.

I am not certain it was a good choice, but I want to I2cWrapper and SpiWrapper do remain consistent.

Another thing, I have been wondering whether these two wrapper classes belonged in the `mod.rs` file at all, or whether we should think about breaking it up into smaller files, it has grown significantly, even just the `TransportWrapper` itself, and its separate builder class.

Maybe we can think of a followup cleanup CL, dealing both with member naming, and file structure.

Reply:
Agreed.